tool/tsh/common/db.go

```diff
@@ -863,6 +863,23 @@ func newDatabaseInfo(cf *CLIConf, tc *client.TeleportClient, route *tlsca.RouteT
 // checkAndSetPrincipalDefaults checks the db route (schema) name and username,
 // and sets them to defaults if necessary.
 func (d *databaseInfo) checkAndSetPrincipalDefaults(cf *CLIConf, tc *client.TeleportClient, db types.Database) error {
+	profile, err := tc.ProfileStatus()
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	// if either user or db name isn't given as a cli flag, try to populate
+	// user/db name from an active db cert.
+	if cf.DatabaseUser == "" || cf.DatabaseName == "" {
+		routes, err := profile.DatabasesForCluster(tc.SiteName)
+		if err != nil {
+			return trace.Wrap(err)
+		}
+		if route, ok := findActiveDatabase(d.ServiceName, routes); ok {
+			d.Username = route.Username
+			d.Database = route.Database
+		}
+	}
```

Comment:
Can we cover this flow in `TestTshDB/testDatabaseInfo`  ?

Reply:
[a7c405b](https://github.com/gravitational/teleport/pull/31188/commits/a7c405b792cf630bc18ef585b00401ed8ee10a0e)